LSPS0/README.md

```diff
@@ -0,0 +1,456 @@
+> additional dependencies.
+
+For LSPS0, the message ID SHALL be 37913 (`lsps0_message_id`).
+All LSPS messages use only this message ID.
+
+> **Rationale** We indicate a single message ID as this reduces
+> the "footprint" of all LSPS specifications to only that single
+> message ID, increasing the probability that other protocols
+> using the Lightning BOLT8 peer-to-peer transport will not
+> conflict with the LSPS specifications.
+> The message ID 37913 is odd in order to comply with the "it's
+> OK to be odd" rule, and is in the 32768 to 65535 range reserved
+> for extensions of the BOLT protocol.
+> Some implementations, such as Core Lightning, only expose
+> odd-numbered messages to custom message handlers.
+> The message ID was otherwise selected by random and has no
+> special meaning.
+
+The payload contains the UTF-8 encoding of a complete JSON
+object.
+
+If a client or LSP receives a BOLT8 message with message ID
+37913:
+
+* If the payload does not parse as a complete JSON object
+  (i.e. a JSON key-value store or dictionary), MUST disconnect
+  from the peer and ignore the message.
+* If the payload parses as more than a single complete JSON
+  object, MUST disconnect from the peer and ignore the message.
+* If any byte in the payload is 0, MUST disconnect from the
+  peer and ignore the message.
+* MUST NOT expect the payload to be terminated by a NUL
+  character (0 byte).
+
+> **Rationale** Requiring a single complete JSON object
+> simplifies handling of messages, so that a single message
+> maps to a single request or response.
+> C uses the NUL character as a string terminator and thus
+> embedded 0 bytes may cause problems in implementations
+> that pass the JSON string to C code.
+> Conversely, we do not require the payload to be terminated
+> by a 0 byte / NUL character as it is unnecessary in many
+> modern non-C-based languages; C code can copy the buffer
+> and append the NUL character if necessary, as the payload
+> size is known.
+
+The JSON object embedded in the message payload is defined
+by the [JSON-RPC 2.0][] protocol.
+
+[JSON-RPC 2.0]: https://www.jsonrpc.org/specification
+
+The LSP acts as a JSON-RPC 2.0 server, while the client acts
+as a JSON-RPC 2.0 client.
+
+> **Rationale** JSON is a simple format that is easy to
+> extend and is extensively supported.
+> The Lightning Network peer-to-peer transport protocol in
+> BOLT8 is not inherently a request-response protocol like
+> HTTP(S) or gRPC, and JSON-RPC 2.0 describes a
+> JSON-based protocol that builds a request-response
+> protocol from a simple tunnel; message ID 37913 acts as
+> that tunnel.
+> JSON-RPC 2.0 is simple to implement.
+> Although BOLT8 messages are limited to payloads of 65533
+> bytes, it is expected that both requests and responses
+> would be far below that limit, and thus the ability to
+> "cut" a large object across multiple messages, the
+> use of a compression algorithm, and the use of a binary
```

Comment:
I am wary of adding dependencies at all, but *shrug*. Anyone else want to chime in?

You quoted a "large" response before as 1,946 chars, which UTF-8 encoded will generally be the same size in bytes, which is an order of magnitude below the max size, and then some.  And I believe you quoted 884 bytes when the same response is gzipped.  Now halving the size is good, but how often do we expect something sending via this protocol vs. something sending a 1366-byte `update_add_htlc`?  It makes sense to pack the `update_add_htlc` format into a typical MTU of 1400 bytes - overhead.

For non-large requests and responses, if the pre-compressed size is below MTU anyway, compressing does not really buy us that much in effective networking capacity.  Or are mobile env MTUs smaller?  Another is that for sufficiently short strings, gzip is going to be a loss (i.e. larger because of the header and dictionary), and I suspect most requests / responses to be near the level where gzip will be a loss.

On the other hand, if we want to gzip at all in the future, it is much better to get it early than to have to hack it in later.  It would be painful if we had to first try to parse the packet as a gzipped string, and if that fails, try it as a JSON UTF-8-encoded uncompressed string.  And detecting if the other side is capable of gzip reception would be difficult.

Reply:
TheBlueMatt: in general, no. compression scary

Roy: +1 for kiss

Reply:
Sev: No gzip then 👍

So far hum seems to be "no gzip".